packages/engine/src/config/topology.rs

```diff
@@ -87,61 +29,75 @@ impl Default for AxisBoundary {
     }
 }
 
-/// Configuration of the topology relevant to movement and neighbor calculation
-pub struct Config {
-    /// Dimensions of board associated with "width"
-    pub x_bounds: AxisBoundary,
+#[derive(Deserialize)]
+#[serde(transparent)]
+#[repr(transparent)]
+struct Float(#[serde(deserialize_with = "deserialize_float")] f64);
```

Comment:
Could be nice to add a docstring or call it FloatWithInfinity or something to clarify, but appreciate the impl is directly below

Reply:
Actually bad suggestion from my part with `FloatWithInfinity` as the native type has that already. Probably just keep the docstring and leave it as `Float` or something. Bit hard to encapsulate in a type name that you're trying to support the string "Infinity"